env/JET.env

```diff
@@ -218,6 +280,6 @@ elif [[ "${step}" = "fit2obs" ]]; then
 
     export NTHREADS_FIT2OBS=${nth_fit2obs:-1}
     [[ ${NTHREADS_FIT2OBS} -gt ${nth_max} ]] && export NTHREADS_FIT2OBS=${nth_max}
-    export MPIRUN="${launcher} -n ${npe_fit2obs}"
+    export MPIRUN="${launcher} -n ${npe_fit2obs} --mem=0"
```

Comment:
This asks for all of the memory on the node.  When running Fit2Obs on Jet, I encountered out of memory errors.  The alternative is to add a memory requirement in config.resources, but since it seems to be Jet-specific, I thought I would address it here.

Reply:
How many nodes does Fit2Obs need on Jet? My Fit2Obs PR that went in yesterday has a check for WCOSS2 in `config.resources` so adding a check for Jet to address memory needs is fine. Whichever works best (`--mem=0`) or adding something to `config.resources` is fine. Thanks for explaining what `--mem=0` is doing.

Reply:
The other R&Ds (Hera/Orion) use 3 nodes but they don't have that memory cap so if it needs another node or so on Jet that's fine.

Reply:
When I remove the `--mem=0`, I run into out of memory errors again.  However, `report-mem` is reporting that only ~4GB is being used on each node:

`srun -l --epilog=/apps/local/bin/report-mem --export=ALL -n 3 /lfs1/NESDIS/nesdis-rdo2/David.Huber/Fit2Obs/newm.1.7/exec/cfs_prevmpi /lfs1/NESDIS/nesdis-rdo2/David.Huber/para/stmp/RUNDIRS/ur_192/fit2obs.35214/prec`
...
```
0:  mpicast - background           1
0:  suterps           4         768         384         127
1:  suterps           4         768         384         127
2:  suterps           4         768         384         127
0:  mpicast - background           2
2: k300:UCM:10c7c:73c03540: 7393885 us(7393885 us!!!):  DAPL ERR reg_mr Cannot allocate memory
2: slurmstepd: error: Detected 1 oom-kill event(s) in StepId=25981927.1. Some of your processes may have been killed by the cgroup out-of-memory handler.
srun: error: k300: task 2: Out Of Memory
srun: launch/slurm: _step_signal: Terminating StepId=25981927.1
0: slurmstepd: error: *** STEP 25981927.1 ON k65 CANCELLED AT 2023-04-13T18:23:49 ***
1: forrtl: error (78): process killed (SIGTERM)
1: Image              PC                Routine            Line        Source
1: cfs_prevmpi        0000000000A64FDE  Unknown               Unknown  Unknown
1: libpthread-2.17.s  00002B4319304630  Unknown               Unknown  Unknown
1: libmpi.so.12       00002B4318390C9E  PMPIDI_CH3I_Progr     Unknown  Unknown
1: libmpi.so.12.0     00002B43184F7D5C  Unknown               Unknown  Unknown
1: libmpi.so.12.0     00002B431836E90F  Unknown               Unknown  Unknown
1: libmpi.so.12.0     00002B4318376705  Unknown               Unknown  Unknown
1: libmpi.so.12.0     00002B431837BB67  Unknown               Unknown  Unknown
1: libmpi.so.12       00002B431837B1B3  MPI_Bcast             Unknown  Unknown
1: libmpifort.so.12.  00002B4317F09924  pmpi_bcast            Unknown  Unknown
1: cfs_prevmpi        0000000000427C91  moifun_                   105  moifun.f
1: cfs_prevmpi        000000000040751E  gblevents_                605  gblevn00.f
1: cfs_prevmpi        000000000042A29A  MAIN__                    216  prevents.f
1: cfs_prevmpi        000000000040651E  Unknown               Unknown  Unknown
1: libc-2.17.so       00002B4319835555  __libc_start_main     Unknown  Unknown
1: cfs_prevmpi        0000000000406429  Unknown               Unknown  Unknown
0: forrtl: error (78): process killed (SIGTERM)
0: Image              PC                Routine            Line        Source
0: cfs_prevmpi        0000000000A64FDE  Unknown               Unknown  Unknown
0: libpthread-2.17.s  00002AFC2C555630  Unknown               Unknown  Unknown
0: libmpi.so.12       00002AFC2B5E1C9E  PMPIDI_CH3I_Progr     Unknown  Unknown
0: libmpi.so.12.0     00002AFC2B7491E0  Unknown               Unknown  Unknown
0: libmpi.so.12.0     00002AFC2B5BFB68  Unknown               Unknown  Unknown
0: libmpi.so.12.0     00002AFC2B5C7705  Unknown               Unknown  Unknown
0: libmpi.so.12.0     00002AFC2B5CCB67  Unknown               Unknown  Unknown
0: libmpi.so.12       00002AFC2B5CC1B3  MPI_Bcast             Unknown  Unknown
0: libmpifort.so.12.  00002AFC2B15A924  pmpi_bcast            Unknown  Unknown
0: cfs_prevmpi        0000000000427D0B  moifun_                    74  moifun.f
0: cfs_prevmpi        000000000040751E  gblevents_                605  gblevn00.f
0: cfs_prevmpi        000000000042A29A  MAIN__                    216  prevents.f
0: cfs_prevmpi        000000000040651E  Unknown               Unknown  Unknown
0: libc-2.17.so       00002AFC2CA86555  __libc_start_main     Unknown  Unknown
0: cfs_prevmpi        0000000000406429  Unknown               Unknown  Unknown
srun: error: k65: task 0: Exited with exit code 1
srun: error: k299: task 1: Exited with exit code 1
Peak memory usage summary:
  min =    2412828  KB
  ave =    2838329  KB
  max =    3681276  KB

All nodes sorted by peak memory as percentage of limit: (in KB)
          % of      user        user          user         total       total
Node      limit     max         limit        current      current      phys
k65        3.9     3681276     93716480      2172800      4017816     97584348
k299       2.6     2420884     93716480       318364      2703548     97584348
k300       2.6     2412828     93716480        70236      2661064     97577944
```
So it seems like memory is being assigned by task count rather than by node count.  I'm not sure that adding another node will help.  I've asked the help desk how memory allocations are done by default on Jet.

Reply:
> Would it be possible to rerun one of your Fit2Obs jobs with report-mem in the epilog and let me know the output?

Absolutely...here is the gdasfit2obs job rerun on Hera with `report-mem`:

```
Peak memory usage summary:
  min =    1579108  KB
  ave =    1907166  KB
  max =    2464088  KB

All nodes sorted by peak memory as percentage of limit: (in KB)
          % of      user        user          user         total       total
Node      limit     max         limit        current      current      phys
h17c03     2.6     2464088     94208000       751680      7783884     97470512 
h17c09     1.8     1678304     94208000       775468      8144000     97470516 
h20c48     1.7     1579108     94208000       682940      7067328     97470520
```

/scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/testfit2obs/logs/2022010612/gdasfit2obs.log

I'm thinking we should likely include a memory request for this job now. I probably should have included it in my prior PR.

Reply:
It's seems odd that the memory usage is so different on Hera and Jet (~5.5GB vs ~15GB).  Discounting that, the highest memory usage I saw on Jet out of 51 cases was 18GB, so I'll throw out a memory request of 20GB.

Reply:
I removed `--mem=0` from JET.env and added 20GB request to config.resources.  Testing on the biggest memory hog case was successful.